### DISCUSSION TITLE
Does Supabase offer or intend to offer a caching solution?

### DISCUSSION THREAD

**Discussion by gabedealmeida** [supabase/supabase#2082] [Questions] [answered]
Hey team!

I have a few quick questions.

Does Supabase offer or intend to offer any caching solution for queries to the DB? Does Supabase offer caching for static assets hosted on S3? Thanks, team!

Very best,
Gabriel

**Comment by inian** [comment, marked as answer]
Hey Gabriel, yup. Integrating with a CDN is definitely on the roadmap - probably a few months away though.

**Reply by gabedealmeida** [reply]
Awesome!! How about caching for DB queries?

**Reply by inian** [reply]
Hmm, havent thought about DB queries before. Would allowing you to set a simple `cache-control` header suffice so that it is cached on the client? Or are you thinking of a CDN there too?

**Reply by kellenmace** [reply]
Or as an alternative to cache-control headers, you could wrap requests to Supabase in something like [React Query](https://react-query.tanstack.com/overview) and let it handle client-side caching, since that library works with any API that's Promise-based.
